config.tpl.yml

```diff
@@ -30,9 +30,18 @@ APPS:
     min_instances: {{ MIN_INSTANCE_COUNT_API }}
     max_instances: {{ MAX_INSTANCE_COUNT_API }}
     scalers:
-      - type: ElbScaler
-        elb_name: 'notify-paas-proxy'
-        threshold: 600
+      - type: ScheduleScaler
+        schedule:
+          scale_factor: 0.6
+          workdays:
+            - 08:00-23:00
+          weekends:
+            - 08:00-23:00
+
+  - name: notify-api-sms-callbacks
+    min_instances: {{ MIN_INSTANCE_COUNT_HIGH }}
+    max_instances: {{ MAX_INSTANCE_COUNT_MEDIUM }}
+    scalers:
       - type: ScheduleScaler
```

Comment:
So I'm looking at the schedule scaler code and this rings a bell from before because I think I had a similar conversation with Leo in the past about how it doesn't do what it looks like it should do and it's very confusing.

https://github.com/alphagov/notifications-paas-autoscaler/blob/master/app/schedule_scaler.py#L10

From that code it looks like during workday and weekend hours, the number of instances will be set by

```return int(math.ceil(self.max_instances * self.scale_factor))```

which for us would be 10*0.6 = 6.

Then not during workhours it would be 
```
if not self._should_scale_on_schedule():
            return self.min_instances
```

and our min instances is 4.

So I think this will set it as 4 instances overnight, 6 instances during the day.

If we want it to be 4 instances overnight and 10 during the day, we would want to change the scale factor to `1` I believe.

Please double check this all please as writing this quickly before I go for lunch!

Reply:
Yes, you're right, the scaling factor should be 1 if we want to scale all the way up to `max_instances`.

As for the `ScheduleScaler` for `notify-api`, given that `MIN_INSTANCE_COUNT_API == MAX_INSTANCE_COUNT_API` it will essentially be a no-op and will remain at `MIN_INSTANCE_COUNT_API` instances.

Reply:
However, if you look at production, notify-api is not scaling down in the evenings.

Reply:
Yes, because it will never drop below the minimum instances:

https://github.com/alphagov/notifications-paas-autoscaler/blob/2191485e445d262092dbe660aed33df535a27592/app/base_scalers.py#L22

Reply:
Api doesn't scale down in the evenings because min and max instances are set the same, https://github.com/alphagov/notifications-paas-autoscaler/blob/2191485e445d262092dbe660aed33df535a27592/app/base_scalers.py#L20-L23